.jenkins/common.groovy

```diff
@@ -35,11 +35,11 @@ def runCompileCommand(platform, project, jobName, boolean debug=false)
     // avoid bug causing long build times of certain files.
     String buildType = 'Release' // debug ? 'Debug' : 'RelWithDebInfo'
     String parallelJobs = "export HIPCC_COMPILE_FLAGS_APPEND='-O3 -Wno-format-nonliteral -parallel-jobs=4'"
-    String buildThreads = '-1' // if hipcc is used may be multiplied by parallel-jobs
-    if (platform.jenkinsLabel.contains('gfx11'))
-        buildThreads = '16'
-
-    // comment
+    
+    def systemCPUs = sh(script: 'nproc', returnStdout: true ).trim().toInteger()
+    def systemRAM = sh(script: 'free -g | grep -P "[[:digit:]]+" -m 1 -o | head -n 1', returnStdout: true ).trim().toInteger()
+    def maxThreads = min(min(systemCPUs, systemRAM / 10), 64)
```

Comment:
Maybe want a max(1, ...) and systemRAM / 10 , you don't need 10GB per thread do you?  / 2 ?  Otherwise fine builds on all pipes.

Reply:
Good idea for max(1, ...), I'll add that in the next update.
Not sure if 10GB is needed, but latest gfx11 build still crashed, so may need to reduce threading on that platform further. My code checks total available memory, but maybe I should update it to check available memory to account for other jobs running? But then threading selected may vary widely from run to run.
Also debugging some other unrelated errors that are appearing in CI builds, appear to be caused by CI's update to the latest docker.